src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -1779,13 +1779,20 @@ fn prepare_enum_metadata(
                 .discriminants(cx.tcx)
                 .zip(&def.variants)
                 .map(|((_, discr), v)| {
-                    let name = SmallCStr::new(&v.ident.as_str());
+                    let name = v.ident.as_str();
```

Comment:
```suggestion
                    let name: &'static str = v.ident.as_str().string;
```

`SmallCStr` is an owning type. Does `DIBuilderCreateEnumerator` take ownership of the name pointer? `as_str` returns a reference to the symbol interner, so this a use-after-free unless the interner is guaranteed to outlive our calls to LLVM. Is this the case? If we need to do away with the calls to `SmallCStr`, we should at least ensure that the lifetime of the returned `str` is indeed `static`.

Reply:
DIBuilderCreateEnumerator borrows the name for the duration of the call, there
is no transfer of ownership. SmallCStr was used to signal string length through
zero termination, which is now done through separate argument (which avoids
creating a temporary copy).